### PR TITLE
refactor(msg_type): move it to its own file

### PR DESCRIPTION
**Pull request by oetyng**
refactor(msg_type): move it to its own file

- Moves priority fns to service- and system msg.
- Moves deserialise of payload to wire_msg fn when getting priority.

### CODE REVIEW

File: sn_client/src/connections/messaging.rs [hunk 1]
```diff
@@ -104,7 +104,7 @@ impl Session {
         };
 
         let msg_kind = AuthKind::Service(auth);
-        let wire_msg = WireMsg::new_msg(msg_id, payload, msg_kind, dst_location)?;
+        let wire_msg = WireMsg::new_msg(msg_id, payload, msg_kind, 0, dst_location)?;
```
Comment (joshuef) [inline sn_client/src/connections/messaging.rs:107]: Q @oetyng: can this open the door to an attack? Can clients not just put the prio as they want? (and risk getting blacklisted perhaps? , but better if that could be avoided altogether...)

Comment (oetyng) [reply sn_client/src/connections/messaging.rs:107]: Ah you're right, that's not handled here. There's a fix that I can see immediately, but I'm trying to first remember what I had in mind when I did this (~3 months ago).. also not sure I really need this for the cmd ctrl PR..  *thinking*

Comment (oetyng) [reply sn_client/src/connections/messaging.rs]: Ok, I solved it by changing the PR main focus :) 
So, I've skipped that change.

**Review by joshuef — APPROVED** [review]

**Pull request merged**